tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -61,16 +129,153 @@ def startsWith(substring):
     """Validate that string value isn't only blanks."""
     return make_validator(
-        lambda value: not value[start:end if end else len(value)].isspace(),
-        lambda value: f'{value} contains blanks between positions {start} and {end if end else len(value)}.'
+        lambda value: not str(value)[start:end if end else len(str(value))].isspace(),
+        lambda value: f'{str(value)} contains blanks between positions {start} and {end if end else len(str(value))}.'
     )
 
+def isEmpty(start=0, end=None):
+    """Validate that string value is only blanks."""
+    return make_validator(
+        lambda value: value[start:end if end else len(value)].isspace(),
+        lambda value: f'{value} is not blank between positions {start} and {end if end else len(value)}.'
+    )
+
+
+def notZero(number_of_zeros=1):
+    """Validate that value is not zero."""
+    return make_validator(
+        lambda value: value != '0' * number_of_zeros,
+        lambda value: f'{value} is zero.'
+    )
+
+
+def isLargerThan(LowerBound):
+    """Validate that value is larger than the given value."""
+    return make_validator(
+        lambda value: float(value) > LowerBound if value is not None else False,
+        lambda value: f'{value} is not larger than {LowerBound}.'
+    )
+
+
+def isSmallerThan(UpperBound):
+    """Validate that value is smaller than the given value."""
+    return make_validator(
+        lambda value: value < UpperBound,
+        lambda value: f'{value} is not smaller than {UpperBound}.'
+    )
+
+def isLargerThanOrEqualTo(LowerBound):
+    """Validate that value is larger than the given value."""
+    return make_validator(
+        lambda value: value >= LowerBound,
+        lambda value: f'{value} is not larger than {LowerBound}.'
+    )
+
+def isSmallerThanOrEqualTo(UpperBound):
+    """Validate that value is smaller than the given value."""
+    return make_validator(
+        lambda value: value <= UpperBound,
+        lambda value: f'{value} is not smaller than {UpperBound}.'
+    )
+
+def isInLimits(LowerBound, UpperBound):
+    """Validate that value is in a range including the limits."""
+    return make_validator(
+        lambda value: value >= LowerBound and value <= UpperBound,
+        lambda value: f'{value} is not larger and equal to {LowerBound} and smaller and equal to {UpperBound}.'
+    )
 
 # custom validators
 
+def month_year_monthIsValid():
+    """Validate that in a monthyear combination, the month is a valid month."""
+    return make_validator(
+        lambda value: int(str(value)[4:6]) in range(1, 13),
+        lambda value: f'{str(value)[4:6]} is not a valid month.'
+    )
+
+def olderThan(min_age):
+    """Validate that value is larger than min_age."""
+    return make_validator(
+        lambda value: date.today().year - int(str(value)[:4]) > min_age,
+        lambda value: f'{date.today().year - int(str(value)[:4])} is not larger than {min_age}.'
+    )
+
+def month_year_yearIsLargerThan(year):
```

Comment:
Can we improve these function names with the following forms?
`dateMonthIsValid()`, `dateOlderThan()`, and `dateYearIsLargerThan()`?

Reply:
Do you want that change made here or in the 2369 PR? This function is from 2369.

Reply:
My apologies. Please do make the change here as 2369 is deep into qasp and alex did not want any new commits. I'm just not a fan of combining underscores and camelCase.